todo-list/src/App.js

```diff
@@ -0,0 +1,107 @@
+import React, { useState } from 'react';
+import logo from './logo.svg';
+import './App.css';
+import TodoForm from './components/TodoForm';
+import TodoList from './components/TodoList';
+
+function App() {
+	const [todo, setTodo] = useState([]);
+	const [inProgress, setInProgress] = useState([]);
+	const [done, setDone] = useState([]);
+	const [addNew, setAddNew] = useState(false);
+
+	const addNewTask = (task) => {
+		switch (task.status) {
+			case 'todo':
+				setTodo((todo) => [...todo, task]);
+				setAddNew(false);
+				break;
+			case 'inProgress':
+				setInProgress((inProgress) => [...inProgress, task]);
+				setAddNew(false);
+
+				break;
+			case 'done':
+				setDone((done) => [...done, task]);
+				setAddNew(false);
+
+				break;
+		}
+	};
+
+	const removeTask = (task, index) => {
+		switch (task.status) {
+			case 'todo':
+				const todoCopy = [...todo];
+				todoCopy.splice(index, 1);
+				setTodo(todoCopy);
+				break;
+			case 'inProgress':
+				const inProgressCopy = [...inProgress];
+				inProgressCopy.splice(index, 1);
+				setInProgress(inProgressCopy);
+				break;
+			case 'done':
+				const doneCopy = [...done];
+				doneCopy.splice(index, 1);
+				setDone(doneCopy);
+				break;
+		}
+	};
+	const moveTask = (action, task, index) => {
+		switch (action) {
+			case 'todo':
+				removeTask(task, index);
+				task.status = 'todo';
+				setTodo((todo) => [...todo, task]);
+				break;
+			case 'inProgress':
+				removeTask(task, index);
+				task.status = 'inProgress';
+				setInProgress((inProgress) => [...inProgress, task]);
+				break;
+			case 'done':
+				removeTask(task, index);
+				task.status = 'done';
+				setDone((done) => [...done, task]);
+				break;
+		}
+	};
+
+	return (
+		<div className="App">
+			{addNew ? (
+				<TodoForm addNewTask={addNewTask} />
+			) : (
+				<button className="button" onClick={() => setAddNew(true)}>
+					{' '}
+					add new task{' '}
+				</button>
+			)}
+
+			<div class="columns">
+				<div class="column">
+					{' '}
+					<h1>to do</h1>
+					<TodoList tasks={todo} removeTask={removeTask} moveTask={moveTask} />
+				</div>
+				<div class="column">
+					{' '}
+					<h1>in progress</h1>
+					<TodoList
+						tasks={inProgress}
+						removeTask={removeTask}
+						moveTask={moveTask}
+					/>
+				</div>
+				<div class="column">
+					{' '}
+					<h1>done</h1>
+					<TodoList tasks={done} removeTask={removeTask} moveTask={moveTask} />
+				</div>
```

Comment:
Since these columns are a lot the same you could put the values in an array of objects and then map through them to cut down on lines.

Reply:
Definitely would've been neater! I'll keep that in mind, thank you.